exporter/stackdriverexporter/factory.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackdriverexporter
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter"
+)
+
+type factory struct {
```

Comment:
If you embed the `component.ExporterFactory` instead of naming the field `delegate`, you can avoid writing the delegation logic by hand.

```go
const configType = configmodels.Type("stackdriver")

type factory struct { component.ExporterFactory }

func (*factory) Type() configmodels.Type { return configType }

func (f *factory) CreateDefaultConfig() configmodels.Exporter {
  cfg := f.ExporterFactory.CreateDefaultConfig()
  cfg.(*googlecloudexporter.Config).TypeVal = configType
  return cfg
}

// no need to implement CreateTracesExporter etc - they are inherited
```

Reply:
I wrote the initial example with delegate but this is better :)

Reply:
@punya hmm I gave this a shot but it segfaults (I think) because the `component.ExporterFactory` doesn't implement those `Create*Exporter`. See https://github.com/aabmass/opentelemetry-collector-contrib/commit/e6ebaf74c98e0c8eaf167ce475bde6466e065f43